tests/cli.test.ts

```diff
@@ -0,0 +1,215 @@
+
+import Mux from '@mux/mux-node';
+import yargs from 'yargs';
+import log from '../src/logger.js';
+
+import { handler, builder } from '../src/cli/sync.js';
+import { createAsset, updateAsset } from '../src/assets.js';
+
+import * as fakeMux from './utils/fake-mux.js';
+
+function findConsoleMessage(consoleSpy: any, regex: RegExp) {
+  return consoleSpy.mock.calls.find(({ arguments: messages }) => {
+    return messages.join('').match(regex);
+  });
+}
+
+async function createFakeVideoFile(dir: string, filename: string = 'video.mp4'): Promise<string> {
+  const filePath = path.join(dir, filename);
+  await fs.writeFile(filePath, 'fake video data');
+
+  return filePath;
+}
+
+const silenceLog =
+  (silence: boolean = true) =>
+  (type: string, ...messages: string[]) => {
+    if (silence) {
+      return;
+    }
+
+    console.log(type, ...messages);
+  };
+
+// I really hate this, but for whatever reason we're unable to mock the base log function
+// so we have to mock each individual method instead.
+function logSpies(ctx: any, silence: boolean = true) {
+  return {
+    infoSpy: ctx.mock.method(log, 'info', silenceLog(silence)),
+    warningSpy: ctx.mock.method(log, 'warning', silenceLog(silence)),
+    successSpy: ctx.mock.method(log, 'success', silenceLog(silence)),
+    errorSpy: ctx.mock.method(log, 'error', silenceLog(silence)),
+    addSpy: ctx.mock.method(log, 'add', silenceLog(silence)),
+    spaceSpy: ctx.mock.method(log, 'space', silenceLog(silence)),
+  };
+}
+
+describe('cli', () => {
+  let server: http.Server;
+  let tmpDirs: string[] = []; // Keep track of temporary directories so we can clean them up.
+
+  async function createTempDir(): Promise<string> {
+    // Create a temporary directory and populate it with some files.
+    const dir = await fs.mkdtemp(path.join('tests', 'tmp-videos-'));
+
+    tmpDirs.push(dir);
+
+    return dir;
+  }
+
+  before(() => {
+    mock.method(Mux.prototype, 'post', fakeMux.post);
+    mock.method(Mux.prototype, 'get', fakeMux.get);
+
+    server = http.createServer((req, res) => {
+      res.writeHead(200, { 'Content-Type': 'text/plain' });
+      res.end('OK');
+    });
+
+    server.listen(3123, () => {
+      // console.info('Dummy upload server running on http://localhost:3123');
+    });
+  });
+
+  after(() => {
+    server.close();
+
+    tmpDirs.forEach(async (dir) => {
+      await fs.rm(dir, { recursive: true, force: true });
+    });
+  });
+
+  describe('sync', () => {
+    it('logs a warning and bails if the specified `dir` does not exist', async (t) => {
+      const { warningSpy, infoSpy } = logSpies(t);
+
+      const args = builder(yargs('')).parseSync();
+
+      await handler(args);
+
+      assert(findConsoleMessage(warningSpy, /Directory does not exist/i), 'Directory does not exist message not found');
+      assert(
+        findConsoleMessage(infoSpy, /Did you forget to run next-video init/i),
+        'Did you forget to run next-video init message not found'
```

Comment:
Not a blocker, but might make sense to store the text as constants somewhere, so changes to the text don't have to be made in two places.